core/src/main/java/io/camunda/migrator/RuntimeMigrator.java

```diff
@@ -387,53 +394,63 @@ protected void activateMigratorJobs() {
     List<ActivatedJob> migratorJobs = null;
     do {
       var jobQuery = camundaClient.newActivateJobsCommand()
-          .jobType("migrator")
+          .jobType(migratorProperties.getJobActivationType())
           .maxJobsToActivate(getBatchSize());
 
       String fetchMigratorJobsErrorMessage = "Error while fetching migrator jobs";
       migratorJobs = callApi(() -> jobQuery.execute().getJobs(), fetchMigratorJobsErrorMessage);
 
       LOGGER.debug("Migrator jobs found: {}", migratorJobs.size());
 
-      migratorJobs.forEach(activatedJob -> {
-        String fetchLegacyIdErrorMessage =
-            String.format("Error while fetching legacyId for job with key:" + activatedJob.getProcessInstanceKey());
-        String legacyId = (String) callApi(() -> activatedJob.getVariable("legacyId"), fetchLegacyIdErrorMessage);
-        long processInstanceKey = activatedJob.getProcessInstanceKey();
+      migratorJobs.forEach(job -> {
+        String checkExternallyStartedErrorMessage =
+            "Error while checking if process instance was externally started for job with key: " + job.getProcessInstanceKey();
+        boolean externallyStarted = callApi(() -> !job.getVariables().contains("legacyId"), checkExternallyStartedErrorMessage);
```

Comment:
❓ I may be mistaken here, but this is what I understood so far:
- any variable referenced in the feel expression must exist
- the feel expression will probably reference the `legacyId` variable
- hence, the externally started instances should also have the `legacyId` variable, but probably have it set to an empty string or null

In that case, would `!job.getVariables().contains("legacyId")` not also evaluate to false for externally started instances, as they will contain the variable (although it may have an empty value)?

Reply:
> any variable referenced in the feel expression must exist

No. If `legacyId` doesn't exist, the expression `=if legacyId then "migrator" else "noop"` resolves to `noop`. 
Let me add another test case validating this.

Reply:
Ah right. Then I misunderstood the constraint we mention in the docs:
`Variables referenced in FEEL expressions must exist in the process instance context` Can you clarify?